chaos-days/blog/2024-10-14-realistic-benchmarks/index.md

```diff
@@ -0,0 +1,226 @@
+
+The second goal is to use these benchmarks to derive new minimal viable 
+cluster configuration that can handle at least 40 process instances per 
+second, while maintaining low backpressure and low latency.
+
+The third goal is to scale out minimal viable cluster configuration 
+resources linearly and see if the performance scales accordingly.
+
+**TL;DR;**
+
+We used a realistic process model to benchmark our system and derived a new 
+cluster configuration based on previous requirements. 
+
+When we scale this base configuration linearly we see that the performance
+increases significantly more than linearly, while maintaining low 
+backpressure and low latency.
+
+## Chaos Experiment
+
+### Expected
+
+We do expect that we can find a cluster configuration that can handle at 40 
+tasks second to be significantly reduced in resources in relation to our 
+smaller clusters ([G3-S HA Plan](https://accounts.cloud.dev.ultrawombat.com/consoleadmin/clusterplans/0af08654-28ec-413a-8dcf-c5938e828ddd)) since 
+these can process 50-70 tasks per second. 
+
+We also expect that we can scale this base configuration linearly, and that 
+the processing tasks rate to grow initially a bit faster than linearly due to 
+the lower relative overhead, and if we keep expanding further to flatten due 
+to the partition count being a bottleneck.
+
+### Actual
+
+#### Benchmarking a realistic process model
+
+In our previous benchmarks we used a simple process model with a single 
+task with several decision symbols. For the newer benchmarks we wanted to 
+increase significantly the number of symbols and service tasks used as well 
+to make the load more configurable.
+
+![bank-customer-complaint-dispute-handling](bank-customer-complaint-dispute-handling.png)
+
+For this realistic benchmark, we tried to use a situation closer to a real 
+world scenario, where a bank customer complaint dispute handling process is
+handled by several service tasks. Additionally, fraud claim 
+investigation part of the process is unfolded in multiple instances. This 
+can be configured in our instance payload by how many disputes we send to 
+scale up or down the number of multi instances in the fraud claim 
+investigation, which enables to scale the load of each process instance.
+
+Finally, in deploying our benchmark we can define the number of instances 
+started per second. After choosing a uniform payload for all tests we can 
+start benchmarking or different cluster configurations where we measure the 
+performance by how many process instances are completed per second, while 
+maintaining low backpressure (bellow 10%) as to preserve user experience. 
+
+#### Minimal Viable Requirements for our Cluster
+
+For our minimal viable cluster configuration we determined though several 
+metrics on users and our own experience that we need to be able to handle 
+at least 40 process instances per second, or 3.5 million tasks per day.
```

Comment:
:thought_balloon: I'm not sure how to understand this (as a random user). Maybe something more like: "Based on known customer usage, and our own previous experiments, we determined that the new cluster would need to create and complete a baseline of 40 process instances per second, or about 3.5 million tasks per day."